### PR TITLE
CD 162735 Fix multi page tiff to png (Fix for tiff2vips: Page 1 differs from Page 0)

### PR DESCRIPTION
**Pull request by revathi-murali**
## JIRA

[Main Jira Ticket](https://coupadev.atlassian.net/browse/CD-162735)

## Code Reviews

- [x] @sushmanivargi 
- [ ] @priyankatapar 

## Summary of Change

Changes related to converting multipage tiff to png by converting each page to png and then joining individual pngs

### CODE REVIEW

**Comment by sushmanivargi** [comment]
lgtm

File: vips.go [hunk 1]
```diff
@@ -301,6 +302,35 @@ func vipsRead(buf []byte) (*C.VipsImage, ImageType, error) {
 	return vipsReadWithOptions(buf, Options{})
 }
 
+func vipsArrayJoin(imgArrBuff []*Image) (*C.VipsImage, error) {
+	arrayImage := C.vips_array_image_empty()
+
+	// Convert bimg's image array to VipsImageArray
+	for _, bimgImage := range imgArrBuff {
+		var inImage *C.VipsImage
+		buff := bimgImage.buffer
```
Comment (priyankatapar) [inline vips.go]: https://github.com/coupa/bimg/blob/648549d5b8ae4a78592662c3292383119e0a117c/vips.go#L311-L318

Can we replace this logic with `vipsRead()` method call?

Comment (revathi-murali) [reply vips.go]: Done

**Comment by priyankatapar** [comment]
👍

File: vips.go [hunk 2]
```diff
@@ -301,6 +302,31 @@ func vipsRead(buf []byte) (*C.VipsImage, ImageType, error) {
...
+			return nil, err
+		}
+		vipsArrayImage = C.vips_array_image_append(vipsArrayImage, vipsImage)
+	}
+
+	var out *C.VipsImage
+	imgArrLength := C.int(len(imgArr))
+
+	vipsArrImg := C.vips_array_image_get(vipsArrayImage, &imgArrLength)
+	err := C.vips_arrayjoin_bridge(vipsArrImg, &out, imgArrLength)
```
Comment (abdollar) [inline vips.go]: @revathi-murali maybe call this returnCode or ret to avoid someone confusing this with the idiomatic err error object

**Comment by revathi-murali** [comment]
[lockit v4.9.1] merged via Lockit CLI on 2019-07-16 18:15:19.153179 +0530 IST m=+6.901367359

**Pull request merged**